src/Files.App/Data/Models/StatusBanner.cs

```diff
@@ -0,0 +1,165 @@
+		}
+
+		private bool _IsCancelled;
+		public bool IsCancelled
+		{
+			get => _IsCancelled;
+			set => SetProperty(ref _IsCancelled, value);
+		}
+
+		private int _Progress = 0;
+		public int Progress
+		{
+			get => _Progress;
+			set => SetProperty(ref _Progress, value);
+		}
+
+		private bool _IsProgressing;
+		public bool IsProgressing
+		{
+			get => _IsProgressing;
+			set => SetProperty(ref _IsProgressing, value);
+		}
+
+		private ReturnResult _Status;
+		public ReturnResult Status
+		{
+			get => _Status;
+			set => SetProperty(ref _Status, value);
+		}
+
+		public string Title { get; private set; }
+
+		public FileOperationType Operation { get; private set; }
+
+		public string Message { get; private set; }
+
+		public InfoBarSeverity InfoBarSeverity { get; private set; }
+
+		public string? PrimaryButtonText { get; set; }
+
+		public string SecondaryButtonText { get; set; } = "Cancel".GetLocalizedResource();
+
+		public Action? PrimaryButtonClick { get; }
+
+		public ICommand CancelCommand { get; }
+
+		public bool SolutionButtonsVisible { get; } = false;
+
+		public bool CancelButtonVisible
+			=> CancellationTokenSource is not null;
+
+		public CancellationTokenSource? CancellationTokenSource { get; set; }
+
+		public StatusBanner(string message, string title, float progress, ReturnResult status, FileOperationType operation)
+		{
+			_initialProgress = progress;
+			Message = message;
+			Title = title;
+			FullTitle = title;
+			Status = status;
+			Operation = operation;
+
+			CancelCommand = new RelayCommand(CancelOperation);
+
+			switch (Status)
+			{
+				case ReturnResult.InProgress:
+				{
+					IsProgressing = true;
+
+					if (string.IsNullOrWhiteSpace(Title))
+					{
+						Title = Operation switch
+						{
+							FileOperationType.Extract => Title = "ExtractInProgress/Title".GetLocalizedResource(),
+							FileOperationType.Copy    => Title = "CopyInProgress/Title".GetLocalizedResource(),
+							FileOperationType.Move    => Title = "MoveInProgress".GetLocalizedResource(),
+							FileOperationType.Delete  => Title = "DeleteInProgress/Title".GetLocalizedResource(),
+							FileOperationType.Recycle => Title = "RecycleInProgress/Title".GetLocalizedResource(),
+							FileOperationType.Prepare => Title = "PrepareInProgress".GetLocalizedResource(),
+						};
+					}
+
+					FullTitle = $"{Title} ({_initialProgress}%)";
+
+					break;
+				}
+				case ReturnResult.Success:
+				{
+					IsProgressing = false;
+
+					if (string.IsNullOrWhiteSpace(Title) || string.IsNullOrWhiteSpace(Message))
+						throw new NotImplementedException();
+
+					InfoBarSeverity = InfoBarSeverity.Success;
+
+					break;
+				}
+				case ReturnResult.Failed:
+				case ReturnResult.Cancelled:
+				{
+					IsProgressing = false;
+
+					if (string.IsNullOrWhiteSpace(Title) || string.IsNullOrWhiteSpace(Message))
+						throw new NotImplementedException();
+
+					InfoBarSeverity = InfoBarSeverity.Error;
+
+					break;
+				}
```

Comment:
I think we can merge these cases, all that changes is the value assigned to `InfoBarSeverity`